// @lightbend/akkaserverless-javascript-sdk/index.js
/*
 * Copyright 2021 Lightbend Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * The AkkaServerless module.
 *
 * @module akkaserverless
 */

module.exports.AkkaServerless = require('@lightbend/akkaserverless-javascript-sdk/src/akkaserverless');
module.exports.EventSourcedEntity = require('@lightbend/akkaserverless-javascript-sdk/src/event-sourced-entity');
module.exports.ValueEntity = require('@lightbend/akkaserverless-javascript-sdk/src/value-entity');
module.exports.ReplicatedEntity =
  require('@lightbend/akkaserverless-javascript-sdk/src/replicated-entity').ReplicatedEntity;
module.exports.ReplicatedData =
  require('@lightbend/akkaserverless-javascript-sdk/src/replicated-entity').ReplicatedData;
module.exports.Action = require('@lightbend/akkaserverless-javascript-sdk/src/action');
module.exports.Metadata = require('@lightbend/akkaserverless-javascript-sdk/src/metadata');
module.exports.IntegrationTestkit = require('@lightbend/akkaserverless-javascript-sdk/src/integration-testkit');
module.exports.View = require('@lightbend/akkaserverless-javascript-sdk/src/view');
module.exports.replies = require('@lightbend/akkaserverless-javascript-sdk/src/reply');
module.exports.settings = require('@lightbend/akkaserverless-javascript-sdk/settings');
