// @lightbend/akkaserverless-javascript-sdk/proto/akkaserverless/component/eventsourcedentity/event_sourced_entity_pb.js
// source: akkaserverless/component/eventsourcedentity/event_sourced_entity.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var akkaserverless_component_component_pb = require('@lightbend/akkaserverless-javascript-sdk/proto/akkaserverless/component/component_pb');
goog.object.extend(proto, akkaserverless_component_component_pb);
var akkaserverless_component_entity_entity_pb = require('@lightbend/akkaserverless-javascript-sdk/proto/akkaserverless/component/entity/entity_pb');
goog.object.extend(proto, akkaserverless_component_entity_entity_pb);
var google_protobuf_any_pb = require('google-protobuf/google/protobuf/any_pb.js');
goog.object.extend(proto, google_protobuf_any_pb);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent', null, global);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedInit', null, global);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedReply', null, global);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot', null, global);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn', null, global);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.MessageCase', null, global);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut', null, global);
goog.exportSymbol('proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.MessageCase', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.akkaserverless.component.eventsourcedentity.EventSourcedInit, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.displayName = 'proto.akkaserverless.component.eventsourcedentity.EventSourcedInit';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.displayName = 'proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.displayName = 'proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.repeatedFields_, null);
};
goog.inherits(proto.akkaserverless.component.eventsourcedentity.EventSourcedReply, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.displayName = 'proto.akkaserverless.component.eventsourcedentity.EventSourcedReply';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.oneofGroups_);
};
goog.inherits(proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.displayName = 'proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.oneofGroups_);
};
goog.inherits(proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.displayName = 'proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.toObject = function(opt_includeInstance) {
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.toObject = function(includeInstance, msg) {
  var f, obj = {
    serviceName: jspb.Message.getFieldWithDefault(msg, 1, ""),
    entityId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    snapshot: (f = msg.getSnapshot()) && proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.akkaserverless.component.eventsourcedentity.EventSourcedInit;
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setServiceName(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setEntityId(value);
      break;
    case 3:
      var value = new proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot;
      reader.readMessage(value,proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.deserializeBinaryFromReader);
      msg.setSnapshot(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getServiceName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getEntityId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getSnapshot();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.serializeBinaryToWriter
    );
  }
};


/**
 * optional string service_name = 1;
 * @return {string}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.getServiceName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.setServiceName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string entity_id = 2;
 * @return {string}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.getEntityId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.setEntityId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional EventSourcedSnapshot snapshot = 3;
 * @return {?proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.getSnapshot = function() {
  return /** @type{?proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot} */ (
    jspb.Message.getWrapperField(this, proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot, 3));
};


/**
 * @param {?proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.setSnapshot = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.clearSnapshot = function() {
  return this.setSnapshot(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.prototype.hasSnapshot = function() {
  return jspb.Message.getField(this, 3) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.toObject = function(opt_includeInstance) {
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.toObject = function(includeInstance, msg) {
  var f, obj = {
    snapshotSequence: jspb.Message.getFieldWithDefault(msg, 1, 0),
    snapshot: (f = msg.getSnapshot()) && google_protobuf_any_pb.Any.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot;
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setSnapshotSequence(value);
      break;
    case 2:
      var value = new google_protobuf_any_pb.Any;
      reader.readMessage(value,google_protobuf_any_pb.Any.deserializeBinaryFromReader);
      msg.setSnapshot(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSnapshotSequence();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getSnapshot();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      google_protobuf_any_pb.Any.serializeBinaryToWriter
    );
  }
};


/**
 * optional int64 snapshot_sequence = 1;
 * @return {number}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.getSnapshotSequence = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.setSnapshotSequence = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional google.protobuf.Any snapshot = 2;
 * @return {?proto.google.protobuf.Any}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.getSnapshot = function() {
  return /** @type{?proto.google.protobuf.Any} */ (
    jspb.Message.getWrapperField(this, google_protobuf_any_pb.Any, 2));
};


/**
 * @param {?proto.google.protobuf.Any|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.setSnapshot = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.clearSnapshot = function() {
  return this.setSnapshot(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedSnapshot.prototype.hasSnapshot = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.toObject = function(opt_includeInstance) {
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.toObject = function(includeInstance, msg) {
  var f, obj = {
    sequence: jspb.Message.getFieldWithDefault(msg, 1, 0),
    payload: (f = msg.getPayload()) && google_protobuf_any_pb.Any.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent;
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setSequence(value);
      break;
    case 2:
      var value = new google_protobuf_any_pb.Any;
      reader.readMessage(value,google_protobuf_any_pb.Any.deserializeBinaryFromReader);
      msg.setPayload(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSequence();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getPayload();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      google_protobuf_any_pb.Any.serializeBinaryToWriter
    );
  }
};


/**
 * optional int64 sequence = 1;
 * @return {number}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.getSequence = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.setSequence = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional google.protobuf.Any payload = 2;
 * @return {?proto.google.protobuf.Any}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.getPayload = function() {
  return /** @type{?proto.google.protobuf.Any} */ (
    jspb.Message.getWrapperField(this, google_protobuf_any_pb.Any, 2));
};


/**
 * @param {?proto.google.protobuf.Any|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.setPayload = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.clearPayload = function() {
  return this.setPayload(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.prototype.hasPayload = function() {
  return jspb.Message.getField(this, 2) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.repeatedFields_ = [3,4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.toObject = function(opt_includeInstance) {
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.toObject = function(includeInstance, msg) {
  var f, obj = {
    commandId: jspb.Message.getFieldWithDefault(msg, 1, 0),
    clientAction: (f = msg.getClientAction()) && akkaserverless_component_component_pb.ClientAction.toObject(includeInstance, f),
    sideEffectsList: jspb.Message.toObjectList(msg.getSideEffectsList(),
    akkaserverless_component_component_pb.SideEffect.toObject, includeInstance),
    eventsList: jspb.Message.toObjectList(msg.getEventsList(),
    google_protobuf_any_pb.Any.toObject, includeInstance),
    snapshot: (f = msg.getSnapshot()) && google_protobuf_any_pb.Any.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.akkaserverless.component.eventsourcedentity.EventSourcedReply;
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setCommandId(value);
      break;
    case 2:
      var value = new akkaserverless_component_component_pb.ClientAction;
      reader.readMessage(value,akkaserverless_component_component_pb.ClientAction.deserializeBinaryFromReader);
      msg.setClientAction(value);
      break;
    case 3:
      var value = new akkaserverless_component_component_pb.SideEffect;
      reader.readMessage(value,akkaserverless_component_component_pb.SideEffect.deserializeBinaryFromReader);
      msg.addSideEffects(value);
      break;
    case 4:
      var value = new google_protobuf_any_pb.Any;
      reader.readMessage(value,google_protobuf_any_pb.Any.deserializeBinaryFromReader);
      msg.addEvents(value);
      break;
    case 5:
      var value = new google_protobuf_any_pb.Any;
      reader.readMessage(value,google_protobuf_any_pb.Any.deserializeBinaryFromReader);
      msg.setSnapshot(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCommandId();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getClientAction();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      akkaserverless_component_component_pb.ClientAction.serializeBinaryToWriter
    );
  }
  f = message.getSideEffectsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      akkaserverless_component_component_pb.SideEffect.serializeBinaryToWriter
    );
  }
  f = message.getEventsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      google_protobuf_any_pb.Any.serializeBinaryToWriter
    );
  }
  f = message.getSnapshot();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      google_protobuf_any_pb.Any.serializeBinaryToWriter
    );
  }
};


/**
 * optional int64 command_id = 1;
 * @return {number}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.getCommandId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.setCommandId = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional akkaserverless.component.ClientAction client_action = 2;
 * @return {?proto.akkaserverless.component.ClientAction}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.getClientAction = function() {
  return /** @type{?proto.akkaserverless.component.ClientAction} */ (
    jspb.Message.getWrapperField(this, akkaserverless_component_component_pb.ClientAction, 2));
};


/**
 * @param {?proto.akkaserverless.component.ClientAction|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.setClientAction = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.clearClientAction = function() {
  return this.setClientAction(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.hasClientAction = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * repeated akkaserverless.component.SideEffect side_effects = 3;
 * @return {!Array<!proto.akkaserverless.component.SideEffect>}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.getSideEffectsList = function() {
  return /** @type{!Array<!proto.akkaserverless.component.SideEffect>} */ (
    jspb.Message.getRepeatedWrapperField(this, akkaserverless_component_component_pb.SideEffect, 3));
};


/**
 * @param {!Array<!proto.akkaserverless.component.SideEffect>} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.setSideEffectsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.akkaserverless.component.SideEffect=} opt_value
 * @param {number=} opt_index
 * @return {!proto.akkaserverless.component.SideEffect}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.addSideEffects = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.akkaserverless.component.SideEffect, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.clearSideEffectsList = function() {
  return this.setSideEffectsList([]);
};


/**
 * repeated google.protobuf.Any events = 4;
 * @return {!Array<!proto.google.protobuf.Any>}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.getEventsList = function() {
  return /** @type{!Array<!proto.google.protobuf.Any>} */ (
    jspb.Message.getRepeatedWrapperField(this, google_protobuf_any_pb.Any, 4));
};


/**
 * @param {!Array<!proto.google.protobuf.Any>} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.setEventsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.google.protobuf.Any=} opt_value
 * @param {number=} opt_index
 * @return {!proto.google.protobuf.Any}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.addEvents = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.google.protobuf.Any, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.clearEventsList = function() {
  return this.setEventsList([]);
};


/**
 * optional google.protobuf.Any snapshot = 5;
 * @return {?proto.google.protobuf.Any}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.getSnapshot = function() {
  return /** @type{?proto.google.protobuf.Any} */ (
    jspb.Message.getWrapperField(this, google_protobuf_any_pb.Any, 5));
};


/**
 * @param {?proto.google.protobuf.Any|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.setSnapshot = function(value) {
  return jspb.Message.setWrapperField(this, 5, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.clearSnapshot = function() {
  return this.setSnapshot(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.prototype.hasSnapshot = function() {
  return jspb.Message.getField(this, 5) != null;
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.oneofGroups_ = [[1,2,3]];

/**
 * @enum {number}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.MessageCase = {
  MESSAGE_NOT_SET: 0,
  INIT: 1,
  EVENT: 2,
  COMMAND: 3
};

/**
 * @return {proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.MessageCase}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.getMessageCase = function() {
  return /** @type {proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.MessageCase} */(jspb.Message.computeOneofCase(this, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.toObject = function(opt_includeInstance) {
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.toObject = function(includeInstance, msg) {
  var f, obj = {
    init: (f = msg.getInit()) && proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.toObject(includeInstance, f),
    event: (f = msg.getEvent()) && proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.toObject(includeInstance, f),
    command: (f = msg.getCommand()) && akkaserverless_component_entity_entity_pb.Command.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn;
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.akkaserverless.component.eventsourcedentity.EventSourcedInit;
      reader.readMessage(value,proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.deserializeBinaryFromReader);
      msg.setInit(value);
      break;
    case 2:
      var value = new proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent;
      reader.readMessage(value,proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.deserializeBinaryFromReader);
      msg.setEvent(value);
      break;
    case 3:
      var value = new akkaserverless_component_entity_entity_pb.Command;
      reader.readMessage(value,akkaserverless_component_entity_entity_pb.Command.deserializeBinaryFromReader);
      msg.setCommand(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getInit();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.akkaserverless.component.eventsourcedentity.EventSourcedInit.serializeBinaryToWriter
    );
  }
  f = message.getEvent();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent.serializeBinaryToWriter
    );
  }
  f = message.getCommand();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      akkaserverless_component_entity_entity_pb.Command.serializeBinaryToWriter
    );
  }
};


/**
 * optional EventSourcedInit init = 1;
 * @return {?proto.akkaserverless.component.eventsourcedentity.EventSourcedInit}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.getInit = function() {
  return /** @type{?proto.akkaserverless.component.eventsourcedentity.EventSourcedInit} */ (
    jspb.Message.getWrapperField(this, proto.akkaserverless.component.eventsourcedentity.EventSourcedInit, 1));
};


/**
 * @param {?proto.akkaserverless.component.eventsourcedentity.EventSourcedInit|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.setInit = function(value) {
  return jspb.Message.setOneofWrapperField(this, 1, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.clearInit = function() {
  return this.setInit(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.hasInit = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional EventSourcedEvent event = 2;
 * @return {?proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.getEvent = function() {
  return /** @type{?proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent} */ (
    jspb.Message.getWrapperField(this, proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent, 2));
};


/**
 * @param {?proto.akkaserverless.component.eventsourcedentity.EventSourcedEvent|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.setEvent = function(value) {
  return jspb.Message.setOneofWrapperField(this, 2, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.clearEvent = function() {
  return this.setEvent(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.hasEvent = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional akkaserverless.component.entity.Command command = 3;
 * @return {?proto.akkaserverless.component.entity.Command}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.getCommand = function() {
  return /** @type{?proto.akkaserverless.component.entity.Command} */ (
    jspb.Message.getWrapperField(this, akkaserverless_component_entity_entity_pb.Command, 3));
};


/**
 * @param {?proto.akkaserverless.component.entity.Command|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.setCommand = function(value) {
  return jspb.Message.setOneofWrapperField(this, 3, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.clearCommand = function() {
  return this.setCommand(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamIn.prototype.hasCommand = function() {
  return jspb.Message.getField(this, 3) != null;
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.oneofGroups_ = [[1,2]];

/**
 * @enum {number}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.MessageCase = {
  MESSAGE_NOT_SET: 0,
  REPLY: 1,
  FAILURE: 2
};

/**
 * @return {proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.MessageCase}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.getMessageCase = function() {
  return /** @type {proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.MessageCase} */(jspb.Message.computeOneofCase(this, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.toObject = function(opt_includeInstance) {
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.toObject = function(includeInstance, msg) {
  var f, obj = {
    reply: (f = msg.getReply()) && proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.toObject(includeInstance, f),
    failure: (f = msg.getFailure()) && akkaserverless_component_component_pb.Failure.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut;
  return proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.akkaserverless.component.eventsourcedentity.EventSourcedReply;
      reader.readMessage(value,proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.deserializeBinaryFromReader);
      msg.setReply(value);
      break;
    case 2:
      var value = new akkaserverless_component_component_pb.Failure;
      reader.readMessage(value,akkaserverless_component_component_pb.Failure.deserializeBinaryFromReader);
      msg.setFailure(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getReply();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.akkaserverless.component.eventsourcedentity.EventSourcedReply.serializeBinaryToWriter
    );
  }
  f = message.getFailure();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      akkaserverless_component_component_pb.Failure.serializeBinaryToWriter
    );
  }
};


/**
 * optional EventSourcedReply reply = 1;
 * @return {?proto.akkaserverless.component.eventsourcedentity.EventSourcedReply}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.getReply = function() {
  return /** @type{?proto.akkaserverless.component.eventsourcedentity.EventSourcedReply} */ (
    jspb.Message.getWrapperField(this, proto.akkaserverless.component.eventsourcedentity.EventSourcedReply, 1));
};


/**
 * @param {?proto.akkaserverless.component.eventsourcedentity.EventSourcedReply|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.setReply = function(value) {
  return jspb.Message.setOneofWrapperField(this, 1, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.clearReply = function() {
  return this.setReply(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.hasReply = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional akkaserverless.component.Failure failure = 2;
 * @return {?proto.akkaserverless.component.Failure}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.getFailure = function() {
  return /** @type{?proto.akkaserverless.component.Failure} */ (
    jspb.Message.getWrapperField(this, akkaserverless_component_component_pb.Failure, 2));
};


/**
 * @param {?proto.akkaserverless.component.Failure|undefined} value
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut} returns this
*/
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.setFailure = function(value) {
  return jspb.Message.setOneofWrapperField(this, 2, proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut} returns this
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.clearFailure = function() {
  return this.setFailure(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.eventsourcedentity.EventSourcedStreamOut.prototype.hasFailure = function() {
  return jspb.Message.getField(this, 2) != null;
};


goog.object.extend(exports, proto.akkaserverless.component.eventsourcedentity);
