// testcontainers/dist/docker-image-name.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DockerImageName = void 0;
const reaper_1 = require("./reaper");
const port_forwarder_1 = require("./port-forwarder");
class DockerImageName {
    constructor(registry, image, tag) {
        this.registry = registry;
        this.image = image;
        this.tag = tag;
        if (this.registry) {
            this.string = `${this.registry}/${this.image}:${this.tag}`;
        }
        else {
            this.string = `${this.image}:${this.tag}`;
        }
    }
    equals(other) {
        return this.registry === other.registry && this.image === other.image && this.tag === other.tag;
    }
    toString() {
        return this.string;
    }
    isReaper() {
        return this.string === reaper_1.ReaperInstance.getImage();
    }
    isHelperContainer() {
        return DockerImageName.HELPER_CONTAINERS.has(this.string);
    }
    static fromString(string) {
        const registry = this.getRegistry(string);
        const stringWithoutRegistry = registry ? string.split("/").slice(1).join("/") : string;
        if (stringWithoutRegistry.includes("@")) {
            const [image, tag] = stringWithoutRegistry.split("@");
            return new DockerImageName(registry, image, tag);
        }
        else if (stringWithoutRegistry.includes(":")) {
            const [image, tag] = stringWithoutRegistry.split(":");
            return new DockerImageName(registry, image, tag);
        }
        else {
            return new DockerImageName(registry, stringWithoutRegistry, "latest");
        }
    }
    static getRegistry(string) {
        const parts = string.split("/");
        if (parts.length > 1 && this.isRegistry(parts[0])) {
            return parts[0];
        }
    }
    static isRegistry(string) {
        return string.includes(".") || string.includes(":") || string === "localhost";
    }
}
exports.DockerImageName = DockerImageName;
DockerImageName.HELPER_CONTAINERS = new Set([reaper_1.ReaperInstance.getImage(), port_forwarder_1.PortForwarderInstance.getImage()]);
