// @lightbend/akkaserverless-javascript-sdk/dist/proto/akkaserverless/component/entity/entity_pb.js
"use strict";
// source: akkaserverless/component/entity/entity.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();
var akkaserverless_component_component_pb = require('../../../akkaserverless/component/component_pb.js');
goog.object.extend(proto, akkaserverless_component_component_pb);
var google_protobuf_any_pb = require('google-protobuf/google/protobuf/any_pb.js');
goog.object.extend(proto, google_protobuf_any_pb);
goog.exportSymbol('proto.akkaserverless.component.entity.Command', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.akkaserverless.component.entity.Command = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.akkaserverless.component.entity.Command, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.akkaserverless.component.entity.Command.displayName = 'proto.akkaserverless.component.entity.Command';
}
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.akkaserverless.component.entity.Command.prototype.toObject = function (opt_includeInstance) {
        return proto.akkaserverless.component.entity.Command.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.akkaserverless.component.entity.Command} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.akkaserverless.component.entity.Command.toObject = function (includeInstance, msg) {
        var f, obj = {
            entityId: jspb.Message.getFieldWithDefault(msg, 1, ""),
            id: jspb.Message.getFieldWithDefault(msg, 2, 0),
            name: jspb.Message.getFieldWithDefault(msg, 3, ""),
            payload: (f = msg.getPayload()) && google_protobuf_any_pb.Any.toObject(includeInstance, f),
            streamed: jspb.Message.getBooleanFieldWithDefault(msg, 5, false),
            metadata: (f = msg.getMetadata()) && akkaserverless_component_component_pb.Metadata.toObject(includeInstance, f)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.akkaserverless.component.entity.Command}
 */
proto.akkaserverless.component.entity.Command.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.akkaserverless.component.entity.Command;
    return proto.akkaserverless.component.entity.Command.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.akkaserverless.component.entity.Command} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.akkaserverless.component.entity.Command}
 */
proto.akkaserverless.component.entity.Command.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setEntityId(value);
                break;
            case 2:
                var value = /** @type {number} */ (reader.readInt64());
                msg.setId(value);
                break;
            case 3:
                var value = /** @type {string} */ (reader.readString());
                msg.setName(value);
                break;
            case 4:
                var value = new google_protobuf_any_pb.Any;
                reader.readMessage(value, google_protobuf_any_pb.Any.deserializeBinaryFromReader);
                msg.setPayload(value);
                break;
            case 5:
                var value = /** @type {boolean} */ (reader.readBool());
                msg.setStreamed(value);
                break;
            case 6:
                var value = new akkaserverless_component_component_pb.Metadata;
                reader.readMessage(value, akkaserverless_component_component_pb.Metadata.deserializeBinaryFromReader);
                msg.setMetadata(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.akkaserverless.component.entity.Command.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.akkaserverless.component.entity.Command.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.akkaserverless.component.entity.Command} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.akkaserverless.component.entity.Command.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getEntityId();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getId();
    if (f !== 0) {
        writer.writeInt64(2, f);
    }
    f = message.getName();
    if (f.length > 0) {
        writer.writeString(3, f);
    }
    f = message.getPayload();
    if (f != null) {
        writer.writeMessage(4, f, google_protobuf_any_pb.Any.serializeBinaryToWriter);
    }
    f = message.getStreamed();
    if (f) {
        writer.writeBool(5, f);
    }
    f = message.getMetadata();
    if (f != null) {
        writer.writeMessage(6, f, akkaserverless_component_component_pb.Metadata.serializeBinaryToWriter);
    }
};
/**
 * optional string entity_id = 1;
 * @return {string}
 */
proto.akkaserverless.component.entity.Command.prototype.getEntityId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.akkaserverless.component.entity.Command} returns this
 */
proto.akkaserverless.component.entity.Command.prototype.setEntityId = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional int64 id = 2;
 * @return {number}
 */
proto.akkaserverless.component.entity.Command.prototype.getId = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};
/**
 * @param {number} value
 * @return {!proto.akkaserverless.component.entity.Command} returns this
 */
proto.akkaserverless.component.entity.Command.prototype.setId = function (value) {
    return jspb.Message.setProto3IntField(this, 2, value);
};
/**
 * optional string name = 3;
 * @return {string}
 */
proto.akkaserverless.component.entity.Command.prototype.getName = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};
/**
 * @param {string} value
 * @return {!proto.akkaserverless.component.entity.Command} returns this
 */
proto.akkaserverless.component.entity.Command.prototype.setName = function (value) {
    return jspb.Message.setProto3StringField(this, 3, value);
};
/**
 * optional google.protobuf.Any payload = 4;
 * @return {?proto.google.protobuf.Any}
 */
proto.akkaserverless.component.entity.Command.prototype.getPayload = function () {
    return /** @type{?proto.google.protobuf.Any} */ (jspb.Message.getWrapperField(this, google_protobuf_any_pb.Any, 4));
};
/**
 * @param {?proto.google.protobuf.Any|undefined} value
 * @return {!proto.akkaserverless.component.entity.Command} returns this
*/
proto.akkaserverless.component.entity.Command.prototype.setPayload = function (value) {
    return jspb.Message.setWrapperField(this, 4, value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.entity.Command} returns this
 */
proto.akkaserverless.component.entity.Command.prototype.clearPayload = function () {
    return this.setPayload(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.entity.Command.prototype.hasPayload = function () {
    return jspb.Message.getField(this, 4) != null;
};
/**
 * optional bool streamed = 5;
 * @return {boolean}
 */
proto.akkaserverless.component.entity.Command.prototype.getStreamed = function () {
    return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 5, false));
};
/**
 * @param {boolean} value
 * @return {!proto.akkaserverless.component.entity.Command} returns this
 */
proto.akkaserverless.component.entity.Command.prototype.setStreamed = function (value) {
    return jspb.Message.setProto3BooleanField(this, 5, value);
};
/**
 * optional akkaserverless.component.Metadata metadata = 6;
 * @return {?proto.akkaserverless.component.Metadata}
 */
proto.akkaserverless.component.entity.Command.prototype.getMetadata = function () {
    return /** @type{?proto.akkaserverless.component.Metadata} */ (jspb.Message.getWrapperField(this, akkaserverless_component_component_pb.Metadata, 6));
};
/**
 * @param {?proto.akkaserverless.component.Metadata|undefined} value
 * @return {!proto.akkaserverless.component.entity.Command} returns this
*/
proto.akkaserverless.component.entity.Command.prototype.setMetadata = function (value) {
    return jspb.Message.setWrapperField(this, 6, value);
};
/**
 * Clears the message field making it undefined.
 * @return {!proto.akkaserverless.component.entity.Command} returns this
 */
proto.akkaserverless.component.entity.Command.prototype.clearMetadata = function () {
    return this.setMetadata(undefined);
};
/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.akkaserverless.component.entity.Command.prototype.hasMetadata = function () {
    return jspb.Message.getField(this, 6) != null;
};
goog.object.extend(exports, proto.akkaserverless.component.entity);
